students/ahwall/Lesson05Homework/MailRoomPt3.py

```diff
@@ -0,0 +1,151 @@
+
+donors = [("Donor A", [10, 20, 30, 40]),
+          ("Donor B", [50, 60, 70, 80]),
+          ("Donor C", [90, 100, 110, 120])]
+
+donordict = {'Donor': ['Andrew Wall', 'Jeff Bezos', 'Bill Gates'],
+             'Donation': [100, 260, 420]}
+
+
+def thank_you():
+    ty_letter = input("Please enter the full name of the donor\n\n"
+                      "\tOr type 'list' to view donor name list:  ")
+    if ty_letter == 'list':  # Works
+        print("\nDonor List")
+        print("----------")
+        [print("{: ^10s}".format(items[0])) for items in donors]
+        # loops through to print All Donors in Donor List
+        print()
+        thank_you()
+    else:
+        i = 0
+        while i < 1:
+            for item in donors:
+                DonorInList = False
+                if item[0].count(ty_letter) == 1:
+                    DonorInList = True  # works
+                    break
+            if DonorInList is True:
+                j = 0
+                for item in donors:
+                    if ty_letter == item[0]:  # Works
+                        try:
+                            DonationVal = input("\nDonor found,\n\nHow much was \
+their donate?  ")
+                            donorprof = donors[j]
+                            vallist = donorprof[1]
+                            DonationVal = int(DonationVal)
+                        except ValueError:
+                            print("\nYou did not enter an numberic value, Please \
+try again\n")
+                            thank_you()
+                        else:
+                            vallist.append(DonationVal)
+                            print()
+                            print(ty_letter, "\b's donation has been added to \
+the Donor Report...Creating 'Thank you email'")
+                            i = 1
+                            One_letter(ty_letter, DonationVal)
+                            mainloop()
+                            break
+                    else:
+                        j += 1
+            else:  # WORKS
+                try:
+                    DonationVal = input("\nAdding Donor to list,\n\nHow much was \
+their donate?  ")
+                #  place exception to make sure int is given
+                    DonationVal = int(DonationVal)
+                    vallist = []
+                    vallist.append(DonationVal)
+                except ValueError:
+                    print("\nYou did not enter an numberic value, Please try \
+again\n")
+                    thank_you()
+                else:
+                    Don_Val = ()
+                    Don_Val = (ty_letter[:], vallist[:])
+                    donors.append(Don_Val)
+                    print()
+                    print(ty_letter, "and their donation have been added to \
+                    the Donor Report...Creating 'Thank you email'")
+                    i = 1
+                    One_letter(ty_letter, DonationVal)
+                    mainloop()
```

Comment:
Note that by calling mainloop() you're building the stack, so you might want to avoid doing this